src/infrastructure/figma/transformers/figma-node-transformer.ts

```diff
@@ -37,33 +38,75 @@ export const transformNodeToAtlassianDesign = ({
 			? mapNodeStatusToDevStatus(node.devStatus)
 			: AtlassianDesignStatus.NONE,
 		type: mapNodeTypeToDesignType(node.type),
-		// TODO: lastUpdated should be determined by the nearest parent node's lastModified time once Figma have implemented lastModified
-		lastUpdated: fileResponseWithNode.lastModified,
-		updateSequenceNumber: getUpdateSequenceNumber(fileResponseWithNode.version),
+		lastUpdated: nodeLastModified.toISOString(),
+		updateSequenceNumber: nodeLastModified.getTime(),
 	};
 };
 
-const findNode = (fileResponseWithNode: FileResponse, nodeId: string): Node => {
-	let targetNode = fileResponseWithNode.document;
+/**
+ * Returns a node with the given ID and a date when it was modified.
+ *
+ * Figma API does not provide `lastModified` for each node but only for top-level nodes.
+ * Therefore, the function uses `lastModified` of the corresponding root node as the most accurate value
+ * for the node.
+ *
+ * @remarks
+ * The function uses depth-first search (DFS) for find a node in `fileResponse` (which contains a tree of nodes).
+ * For better performance, fetch {@link FileResponse} with only required nodes -- avoid fetching and traversing the whole
+ * file.
+ *
+ * @internal
+ * Visible for testing only.
+ */
+export const getNodeAndNodeLastModified = (
```

Comment:
**note:** Here is the primary change of this PR. The remaining changes are related to testing only.

CC: @jv3nk